process/block/metablock.go

```diff
@@ -1688,6 +1688,41 @@ func getTxCount(shardInfo []block.ShardData) uint32 {
 	return txs
 }
 
+// DecodeBlockBodyAndHeader method decodes block body and header from a given byte array
+func (mp *metaProcessor) DecodeBlockBodyAndHeader(dta []byte) (data.BodyHandler, data.HeaderHandler) {
+	if dta == nil {
+		return nil, nil
+	}
+
+	var marshalizedBodyAndHeader data.MarshalizedBodyAndHeader
+
+	err := mp.marshalizer.Unmarshal(&marshalizedBodyAndHeader, dta)
+	if err != nil {
+		log.Debug("DecodeBlockBodyAndHeader.Unmarshal: dta", "error", err.Error())
+		return nil, nil
+	}
+
+	var body block.Body
+
+	err = mp.marshalizer.Unmarshal(&body, marshalizedBodyAndHeader.Body)
+	if err != nil {
+		log.Debug("DecodeBlockBodyAndHeader.Unmarshal: marshalizedBodyAndHeader.Body",
+			"error", err.Error())
+		return nil, nil
+	}
+
+	var metaBlock block.MetaBlock
+
+	err = mp.marshalizer.Unmarshal(&metaBlock, marshalizedBodyAndHeader.Header)
+	if err != nil {
+		log.Debug("DecodeBlockBodyAndHeader.Unmarshal: marshalizedBodyAndHeader.Header",
+			"error", err.Error())
+		return nil, nil
+	}
+
+	return body, &metaBlock
+}
+
 // DecodeBlockBody method decodes block body from a given byte array
 func (mp *metaProcessor) DecodeBlockBody(dta []byte) data.BodyHandler {
```

Comment:
duplicated code, the same in shardprocessor.

Reply:
done